### ISSUE TITLE
In anonymized view-only link, GitHub information is not anonymized

### ISSUE THREAD

**Issue by umhan35** [CenterForOpenScience/osf.io#10717]
## What you did (step by step)

1. [Connect GitHub to a project](https://help.osf.io/article/211-connect-github-to-a-project)
2. Create a view-only link to share the OSF project (check off "Anonymize contributor list for this link (e.g., for blind peer review)")
4. Go to the view-only link

## Where does this happen on the OSF?

Anonymized view-only homepage of a project, e.g., https://osf.io/ga9w8/?view_only=3ec356c101944ec092c6badfbb0fb593

## What you expected

The open button should not appear for people to go to the GitHub link

![image](https://github.com/user-attachments/assets/3c9f85b5-26ca-4876-8e59-2267e9c3460a)

## What actually happened

One can click the open button and find out information about the contributors of the OSF repo

## Potential causes

Related code that may have caused this:

https://github.com/CenterForOpenScience/osf.io/blob/3920a29ff5c92a45229e618931661cc5d0c8a08d/addons/github/static/githubFangornConfig.js#L283-L291

## Suggest a solution

Similar to the hidden GitHub repo (username/repo-name) in a view-only link, as seen below, the Open button should also be hidden.

| Not view-only | View-only |
|--------|--------|
| ![image](https://github.com/user-attachments/assets/8cfb5995-70b9-45c1-9921-bb3afe91a150) | ![image](https://github.com/user-attachments/assets/f1e1dca7-4c0f-4fee-a0f3-bc4dd7c02f2f) | 

## Final words

I think the branch list should also be hidden in a view-only link

**Comment by follhim** [comment]
This seems like a major issue to using Github for OSF, I'm adding a comment here to bring attention to this issue.

**Comment by umhan35** [comment]
@follhim I just reproduced this issue with the view-only link in the description, so it is not fixed.

**Comment by umhan35** [comment]
> when ideally, the link shouldn't be accessible at all. Right?

Right. Because the GitHub username/profile reveals the identity of the OSF contributors.

(BTW, I think the branch list should also be hidden in a view-only link)

**Comment by follhim** [comment]
hi @umhan35 , can you try checking this again? I think they may have fixed it?

**Comment by follhim** [comment]
> hi [@umhan35](https://github.com/umhan35) , can you try checking this again? I think they may have fixed it?

Just kidding, I lied, it still doesn't work (i.e., still not anonymous).

**Comment by santoshbs** [comment]
I was wondering if we can expect a fix for this issue. Thx!
